### PR TITLE
[APG-327] Creating case note on status change

### PR DESCRIPTION
**Pull request by anandelangovan1**
## Release checklist

[Release process documentation](../doc/how-to/perform-a-release.md)

As part of our continuous deployment strategy we must ensure that this work is
ready to be released once merged.

### Pre-merge

- [ ] There are changes required to the Accredited Programmes UI for this change to work...
  - [ ] ... and they been released to production already

### Post-merge

- [ ] [Manually approve](../doc/how-to/perform-a-release.md#releasing-to-the-preprod-environment) release to preprod
- [ ] [Manually approve](../doc/how-to/perform-a-release.md#releasing-to-the-production-environment) release to prod

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsaccreditedprogrammesapi/service/CaseNotesApiService.kt [hunk 1]
```diff
@@ -69,19 +78,20 @@ class CaseNotesApiService(
...
     person: PersonEntity?,
     referral: ReferralEntity,
     referralStatusUpdate: ReferralStatusUpdate,
+    message: String,
   ): String {
-    log.warn("Request received for creating case notes :${referral.id} $referralStatusUpdate")
+    log.info("Request received for creating case notes :${referral.id} $referralStatusUpdate")
     val course = referral.offering.course
     val orgName = organisationRepository.findOrganisationEntityByCode(referral.offering.organisationId)?.name
+    val progDescMessage = "Referral to ${course.name}: ${course.audience} strand at $orgName \n"
```
Comment (pfcairns) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsaccreditedprogrammesapi/service/CaseNotesApiService.kt:90]: Could you please expand this variable name to describe what it represents? 

E.g programmeDescriptionMessage or similar?

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsaccreditedprogrammesapi/service/CaseNotesApiService.kt [hunk 2]
```diff
@@ -69,19 +78,20 @@ class CaseNotesApiService(
     }
   }
 
-  private fun buildMessage(
+  fun buildMessage(
```
Comment (pfcairns) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsaccreditedprogrammesapi/service/CaseNotesApiService.kt:81]: buildMessage is a bit vague : could you please rename the method to specify the type of message being built?

File: src/main/resources/db/migration/V92__custom_case_note_message.sql [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+update referral_status set case_notes_subtype = 'PGM_NT_COMP', case_notes_message='PRISONER_NAME cannot continue the programme. The referral will be closed.' where code = 'DESELECTED';
+update referral_status set case_notes_subtype = 'REF_ON_HLD', case_notes_message='The referral will be paused until PRISONER_NAME is ready to continue.' where code = 'ON_HOLD_REFERRAL_SUBMITTED';
+update referral_status set case_notes_subtype = 'REF_ON_HLD', case_notes_message='The referral will be paused until PRISONER_NAME is ready to continue.' where code = 'ON_HOLD_AWAITING_ASSESSMENT';
+update referral_status set case_notes_subtype = 'REF_ON_HLD', case_notes_message='The assessment has not been completed, but the referral will be paused until PRISONER_NAME is ready to continue.' where code = 'ON_HOLD_ASSESSMENT_STARTED';
+update referral_status set case_notes_subtype = 'SUTPGM_NTRDY', case_notes_message='Suitable but not ready (including via deselected open route)' where code = 'SUITABLE_NOT_READY';
+-- No case notes to be added for REFERRAL_STARTED
+update referral_status set case_notes_subtype = '', case_notes_message='' where code = 'REFERRAL_STARTED';
```
Comment (pfcairns) [inline src/main/resources/db/migration/V92__custom_case_note_message.sql:20]: Do we really want to set text columns to an empty string? Would null not be a better option?

**Review by pfcairns — COMMENTED** [review]
Looks good. Just a couple of very minor comments.

**Comment by pfcairns** [comment]
No worries - just seen it was merged.

File: src/main/resources/db/migration/V92__custom_case_note_message.sql (hunk 3 above)
Comment (anandelangovan1) [reply src/main/resources/db/migration/V92__custom_case_note_message.sql:20]: We don't create a case-note for REFERRAL_STARTED. 
if i set it to null or not populate it, then we have to have a check in the codebase for a particular status. This way its all in the config and follows same pattern.

**Pull request merged**